src/unitxt/processors.py

```diff
@@ -277,3 +279,57 @@ def process_value(self, text: Any) -> Any:
 
         except:
             return 0
+
+
+class InferDictsToBinaryLogprobs(FieldOperator):
+    binary_class_names: list[str]
```

Comment:
This is a bit subtle code ,that is  worth documentation.

If I understand right, the approach taken is to get the model to generate a free-form  textual answer (but also ask it to return the top tokens log prob per position) .  Then we find the first position which has at least one of binary choices in the top tokens list.  Then we estimate the probability from the log_probs.  

The benefit of this approach is that it requires only 1 inference per call and fits the existing mechanism.
An alternative run inference twice, once with a forced "Yes" completion and once with a "No" completion. 

"Is this the above anaser faithful to the context? Yes" <-- take the log prob of the completion 
"Is this the above anaser faithful to the context? No" <-- take the log prob of the completion

Then you can be sure you get both option.

Reply:
I'm not sure what I prefer.  If your approach works in most cases, maybe it's simpler.  However, the model may not return "yes" or "no" anywhere in the answer.

Reply:
Your understanding is correct, here we are adding support for this implementation (which seems to work reasonably well) and if someone wants they could add the alternative one as well.